### ISSUE TITLE
Cookie Session Refresh Occurs on Each Page Navigation

### ISSUE THREAD

**Issue by Lady-vamp** [nextauthjs/next-auth#10032]
### Environment

System:
    OS: macOS 14.2.1
    CPU: (16) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 562.64 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 21.6.2 - ~/.nvm/versions/node/v21.6.2/bin/node
    Yarn: 1.22.21 - ~/.nvm/versions/node/v21.6.2/bin/yarn
    npm: 10.2.4 - ~/.nvm/versions/node/v21.6.2/bin/npm
    pnpm: 8.15.2 - ~/.nvm/versions/node/v21.6.2/bin/pnpm
  Browsers:
    Chrome: 121.0.6167.160
    Safari: 17.2.1
  npmPackages:
    next: ^14.1.0 => 14.1.0 
    next-auth: ^5.0.0-beta.11 => 5.0.0-beta.11 
    react: ^18 => 18.2.0 

### Reproduction URL

local repo...

### Describe the issue

```
import NextAuth from 'next-auth';
import Credentials from 'next-auth/providers/credentials';
import bcrypt from 'bcrypt';
import { z } from 'zod';
import { authConfig } from './auth.config';
import { db } from "@/lib/schema";

import { randomBytes, randomUUID } from 'crypto';
import { findUser } from '@/lib/actions';

export const { auth, signIn, signOut, handlers } = NextAuth({
  ...authConfig,
  session: {
    // Choose how you want to save the user session.
    // The default is `"jwt"`, an encrypted JWT (JWE) stored in the session cookie.
    // If you use an `adapter` however, we default it to `"database"` instead.
    // You can still force a JWT session by explicitly defining `"jwt"`.
    // When using `"database"`, the session cookie will only contain a `sessionToken` value,
    // which is used to look up the session in the database.
    strategy: "jwt",

    // Seconds - How long until an idle session expires and is no longer valid.
    maxAge: 30 * 24 * 60 * 60, // 30 days

    // Seconds - Throttle how frequently to write to database to extend a session.
    // Use it to limit write operations. Set to 0 to always update the database.
    // Note: This option is ignored if using JSON Web Tokens
    updateAge: 24 * 60 * 60, // 24 hours

    // The session token is usually either a random UUID or string, however if you
    // need a more customized session token string, you can define your own generate function.
    generateSessionToken: () => {
      return randomUUID?.() ?? randomBytes(32).toString("hex")
    },
  },
  secret: process.env.AUTH_SECRET,
  providers: [
    Credentials({
      async authorize(credentials) {
        const parsedCredentials = z
          .object({ email: z.string().email(), password: z.string().min(6) })
          .safeParse(credentials);

        if (parsedCredentials.success) {
          const { email, password } = parsedCredentials.data;

          const user = await findUser(email);
          if (!user) return null;

          const passwordsMatch = await bcrypt.compare(password, user.password);

          if (passwordsMatch) return user;
        }

        console.log('Invalid credentials');
        return null;
      },
    })
  ]
});
```

Cookie Session Refresh Occurs on Each Page Navigation, In the server logs I also get this
```
{ x-middleware-next: '1' }
{
  set-cookie: 'authjs.session-token=token...; Path=/; Expires=Sat, 16 Mar 2024 07:59:48 GMT; HttpOnly; SameSite=Lax',
  x-middleware-next: '1'
}
```
basically it keeps refreshing the cookie every page you load. 

### How to reproduce

Just change page and the issue arise

### Expected behavior

Not sure what version I had before but it broke last night since I ran `npm upgrade`
